KIPs/kip-276.md

```diff
@@ -0,0 +1,338 @@
+---
+kip: 276
+title: eth_config JSON-RPC Method
+author: Garry (@Mdaiki0730), Ollie (@blukat29)
+discussions-to: https://devforum.kaia.io/t/discussion-on-kip-276/8822
+status: Draft
+type: Core
+created: 2025-10-15
+---
+
+## Abstract
+
+Add a new api endpoint to fetch the configuration for each fork. This KIP focuses on describing the difference from [EIP-7910](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7910.md).
+
+## Motivation
+
+Increases transparency of configuration values ​​for each fork by adding an `eth_config` api endpoint.
+This API will allow users to check that their node are ready for the next fork and that the current fork configuration values ​​are correct.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+All clients constraints directly under the specification follow [EIP-7910](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7910.md).
+
+### Configuration RPC
+
+A new JSON-RPC API, `eth_config`, is introduced. It takes no parameters and returns the result object specified in the next section.
+
+### Result Object Structure
+
+The RPC result object structure follows [EIP-7910](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7910.md).
+
+### Members of the Configuration Object
+
+The menbers of the configuration object follows that of [EIP-7910](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7910.md) except for the following fields.
+
+#### `activationTime`
+
+The result MUST NOT have any `activationTime` field.
+
+#### `blobSchedule`
+
+The blob configuration parameters for the specific fork will be represented as specified in [EIP-7910](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7910.md). If [EIP-4844](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-4844.md) isn't implemented in the fork, the `blobSchedule` field MUST be `null`.
+
+#### `forkId`
+
+The forkId follows that of [EIP-7910](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7910.md), i.e. the `FORK_HASH` format as specified in [EIP-6122](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-6122.md).
```

Comment:
```suggestion
The `forkId` follows that of [EIP-7910](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7910.md), i.e. the `FORK_HASH` format as specified in [EIP-6122](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-6122.md).
```

Reply:
Fixed in [7a1a26e](https://github.com/kaiachain/kips/pull/76/commits/7a1a26ed4a23d0f53b05e3836848be8a4907d0be). Thanks!